### DISCUSSION TITLE
Kalman filters

### DISCUSSION THREAD

**Discussion by jbone** [online-ml/river#567] [Q&A]
Has anyone around these parts implemented a Kalman filter in the river context (Ie leveraging river stats / interfaces?)

**Comment by MaxHalford** [comment]
I looked into it ~18 months. But it's a vast topic and my brain couldn't process it very well. It would be wonderful to have a Kalman filter for training ARIMA models online. Implementations such as `auto.arima` in R use a Kalman filter, but the exposed API is only for batch training, whereas Kalman filters are meant to be used in an online fashion. 

Basically someone has to go deep on the topic. It's not a trivial topic that I (or anyone else in the current team) can bang out in a weekend.
